WorkflowSwiftUIExperimental/Sources/SwiftUIScreen.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(UIKit)
+
+import SwiftUI
+import Workflow
+import WorkflowUI
+
+public protocol SwiftUIScreen: Screen {
+    associatedtype Content: View
+
+    @ViewBuilder
+    static func makeView(model: ObservableValue<Self>) -> Content
+
+    static var isDuplicate: ((Self, Self) -> Bool)? { get }
+}
+
+public extension SwiftUIScreen {
+    static var isDuplicate: ((Self, Self) -> Bool)? { return nil }
+}
+
+public extension SwiftUIScreen where Self: Equatable {
+    static var isDuplicate: ((Self, Self) -> Bool)? { { $0 == $1 } }
+}
+
+public extension SwiftUIScreen {
+    func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+        ViewControllerDescription(
+            type: ModeledHostingController<Self, WithModel<Self, EnvironmentInjectingView<Content>>>.self,
+            environment: environment,
+            build: {
+                let (model, modelSink) = ObservableValue.makeObservableValue(self, isDuplicate: Self.isDuplicate)
+                let (viewEnvironment, envSink) = ObservableValue.makeObservableValue(environment)
+                return ModeledHostingController(
+                    modelSink: modelSink,
+                    viewEnvironmentSink: envSink,
+                    rootView: WithModel(model, content: { model in
+                        EnvironmentInjectingView(
+                            viewEnvironment: viewEnvironment,
+                            content: Self.makeView(model: model)
+                        )
+                    })
+                )
+            },
+            update: {
+                $0.modelSink.send(self)
+                $0.viewEnvironmentSink.send(environment)
+            }
+        )
+    }
+}
+
+private struct EnvironmentInjectingView<Content: View>: View {
+    @ObservedObject var viewEnvironment: ObservableValue<ViewEnvironment>
+    let content: Content
+
+    var body: some View {
+        content
+            .environment(\.viewEnvironment, viewEnvironment.value)
+    }
+}
+
+private final class ModeledHostingController<Model, Content: View>: UIHostingController<Content> {
```

Comment:
For my own understanding, this is so we can basically isolate each VC into its own SwiftUI hierarchy?

Reply:
I think of it as simply: we need some view controller class for `SwiftUIScreen`'s `viewControllerDescription` to describe, and that view controller must expose an interface for the description to `update` with both the `ViewEnvironment` and screen-specific data.